### ISSUE TITLE
Beaker Test Validation

### ISSUE THREAD

**Issue by Cornellio** [Cornellio/puppet-riak-dd#1]
@danieldreier Hi Daniel, if you wouldn't mind taking a look at this and offering input on running the Beaker tests, as I'm having trouble with them.

I added an augeas resource to set a kernel parameter then added a test to check results via file contents of /etc/sysctl.conf in riak_spec.rb:
`it { is_expected.to contain_file('/etc/sysctl.conf').with_content(/fs.file-max = 65536/) }`

Although the tests pass and the augeas resource is being applied during puppet runs, the tests also pass without the check and pass when I check for a bogus string that should not exist in the file.

The commits are here https://github.com/Cornellio/puppet-riak-dd/commits/riak2_rewrite

thanks

**Comment by danieldreier** [comment]
@Cornellio I'm glad to help. A few things: for beaker tests, you'll want to put the tests in `spec/acceptance/class_spec.rb`. For rspec-puppet tests, it'll be in `spec/classes/riak_spec.rb`.

For something like this that checks the contents of a file, I'd suggest checking it in rspec-puppet rather than a full-on integration tests in beaker, because it'll validate way faster. 

As long as you validate that the right augeas resource is being created, we have a reasonable expectation that the puppet augeas test works, because the augeas type is core puppet and tested heavily by puppet labs. 

That said, If you do want to integration test it in beaker as well, I'd suggest testing the actual settings using the [linux_kernel_parameter](http://serverspec.org/resource_types.html#linux_kernel_parameter) serverspec resource. That would end up integration-testing the whole thing to confirm that the kernel parameter is ultimately getting set, which is the end-goal.

I hope that's helpful - and big +1 for verifying that the check didn't fail when it was expected to.

**Comment by Cornellio** [comment]
Yes that helps. I put the file check in `spec/acceptance/class_spec.rb` and it works when running with `bundle exec rspec spec/acceptance`. I am familiar with running a beaker-rspec test in this way. I'm trying to learn the subtle differences between test types.
